pkg/controller/dependency.go

```diff
@@ -0,0 +1,33 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package controller
+
+import "github.com/talos-systems/os-runtime/pkg/resource"
+
+// DependencyGraph is the exported information about controller/resources dependencies.
+type DependencyGraph struct {
+	Edges []DependencyEdge
+}
+
+// DependencyEdgeType is edge type in controller graph.
+type DependencyEdgeType int
+
+// Controller graph edge types.
+const (
+	EdgeManages DependencyEdgeType = iota
+	EdgeDependsHard
+	EdgeDependsWeak
```

Comment:
Not for this PR but I missed this nit the other day. Strong and Weak go better together naturally. Hard and Soft likewise. I know this aligns with the current convention, but just my 2 cents for the future.

Reply:
thanks, let me update it for this PR until it spreads too far!